client/db/src/lib.rs

```diff
@@ -195,7 +195,7 @@ impl<B: BlockT> StateBackend<HashFor<B>> for RefTrackingState<B> {
 		self.state.for_key_values_with_prefix(prefix, f)
 	}
 
-	fn for_keys_in_child_storage<F: FnMut(&[u8])>(
+	fn for_keys_in_child_storage<F: FnMut(&[u8]) -> bool>(
```

Comment:
existing is rust iterator `for_each`, new is `take_while`. Not sure we if changing name is a good idea, I like the idea of having of having `while` in the name, but I would end with proposing `for_keys_in_child_storage_apply_while` or `in_child_storage_iter_keys_while`  which is way too long for a name :) `.

Reply:
I renamed it to `apply_to_child_keys_while`.